lua/telescope/builtin/internal.lua

```diff
@@ -231,9 +231,35 @@ internal.loclist = function(opts)
 end
 
 internal.oldfiles = function(opts)
-  local results = vim.tbl_filter(function(val)
-    return 0 ~= vim.fn.filereadable(val)
-  end, vim.v.oldfiles)
+  opts.include_current_session = utils.get_default(opts.include_current_session, true)
+
+  local results = {}
+
+  if opts.include_current_session then
+    for _, buffer in ipairs(vim.split(vim.fn.execute(':buffers! t'), "\n")) do
+      local match = tonumber(string.match(buffer, '%s*(%d+)'))
+      if match then
+        local file = vim.api.nvim_buf_get_name(match)
+        if vim.loop.fs_stat(file) and match ~= vim.api.nvim_get_current_buf() then
```

Comment:
you could store current buf in variable outside the loop instead of calling function each time. It's super tiny micro optimization :)

Reply:
Also good call, thanks!